### DISCUSSION TITLE
How to trigger appium tests after github actions?

### DISCUSSION THREAD

**Discussion by MarkYousef1** [appium/appium#19672] [Q&A]
Hi, I am really having difficulties on finding how we trigger appium tests after a github action. I just finished writing my first test with appium for my react native mobile application. Can it be done using github or do we need another service? Please help
Thanks

**Comment by jlipps** [comment]
I have an open source React Native app set up with CI for Appium. My tests run on HeadSpin (not locally on the GitHub Actions VM) but you might be able to use it as inspiration. https://github.com/appium-pro/TheApp/blob/main/CI.md

**Reply by MarkYousef1** [reply]
Can it be done using GitHub actions vm or we necessarily need a service like HeadSpin?

**Reply by saikrishna321** [reply]
@MarkYousef1 You can. Please refer to this yaml as reference https://github.com/AppiumTestDistribution/appium-device-farm/blob/main/azure-pipelines.yml

I have used azure but you can do the same with github actions as well.

**Reply by MarkYousef1** [reply]
@saikrishna321 is it possible to setup a zoom meeting or anything so you can help me setup beacause its still not working

**Reply by saikrishna321** [reply]
CN you share your github actions yaml or the link to the repo?

**Reply by MarkYousef1** [reply]
@saikrishna321 heres the link to my repos:
https://github.com/MarkYousef1/demo.git
This is just for a demo i will later setup the file for my app

**Reply by saikrishna321** [reply]
@MarkYousef1 I see u have success with launching emulator. Now you need to install appium and required drivers and run your test as you run in local.

**Reply by MarkYousef1** [reply]
Hi @saikrishna321 is there a way do it on ubuntu-latest instead of macOS? Please I need help
